In [4]:
import math
from scipy import stats

In [5]:
def perform_hypothesis_testing():
    print("STEP 1: HYPOTHESIS STATEMENT")
    # Theoretical cost model: W = 1000 + 5X
    # For X = 600, theoretical mean cost (mu) = 1000 + 5*(600) = 4000.
    # Note: The problem mixes '$' and 'Rs.', but we treat them as the same underlying cost units.
    print("Null Hypothesis (H0): mu <= 4000 (The weekly operating cost matches the theoretical model)")
    print("Alternative Hypothesis (H1): mu > 4000 (The weekly operating costs are significantly higher)\n")
    # Given Data
    n = 25                      # Sample size (number of restaurants)
    sample_mean = 3050          # Observed sample mean weekly cost (x-bar)
    X_mean = 600                # Mean units produced 
    X_std = 25                  # Standard deviation of units produced
    alpha = 0.05                # Significance level (5%)

    print("STEP 2: CALCULATING THE TEST STATISTIC")
    # 1. Calculate Theoretical Mean Weekly Cost (mu)
    # W = 1000 + 5 * X
    mu = 1000 + (5 * X_mean)
    print(f"Theoretical Mean Weekly Cost (mu): {mu}")
    # 2. Calculate the standard deviation of the cost (sigma)
    # Since W = 1000 + 5X, the constant 1000 does not affect variance. 
    # Therefore, StdDev(W) = 5 * StdDev(X)
    sigma = 5 * X_std
    print(f"Standard Deviation of Cost (sigma): {sigma}")
    # 3. Calculate the Standard Error (SE)
    # SE = sigma / sqrt(n)
    standard_error = sigma / math.sqrt(n)
    print(f"Standard Error (SE): {standard_error}")
    # 4. Calculate the Test Statistic (Z-score or t-score)
    # Formula: (sample_mean - mu) / SE
    # Because we are given the true population parameters for X, we can use the Z-distribution.
    test_statistic = (sample_mean - mu) / standard_error
    print(f"Calculated Test Statistic: {test_statistic:.4f}\n")
    print("STEP 3: DETERMINING THE CRITICAL VALUE")
    # This is a right-tailed (one-tailed) test because owners claim costs are HIGHER.
    # We find the Z-value such that the area to the right is alpha (0.05).
    # stats.norm.ppf(0.95) gives us this critical value.
    critical_value = stats.norm.ppf(1 - alpha)
    print(f"Significance Level (alpha): {alpha}")
    print(f"Critical Value (Z_alpha): {critical_value:.4f}\n")

    print("STEP 4: MAKING A DECISION")
    # For a right-tailed test, we reject H0 if the test statistic is GREATER than the critical value.
    print(f"Is {test_statistic:.4f} > {critical_value:.4f}?")
    
    if test_statistic > critical_value:
        decision = "Reject the Null Hypothesis (H0)."
    else:
        decision = "Fail to Reject the Null Hypothesis (H0)."
        
    print(f"Decision: {decision}\n")

    print("STEP 5: CONCLUSION")
    if "Reject" in decision:
        print("Conclusion: There is insufficient evidence to support the restaurant owners' claim.")
        print("The observed data does not statistically prove that costs have increased beyond the model's prediction.")
    else:
        print("Conclusion: There is strong evidence to support the restaurant owners' claim.")
        print("The weekly operating costs are significantly higher than the original model suggested")
        
        # Note on Data inconsistency:
        print("\n> NOTE: Notice that the sample mean (3,050) is actually LOWER than the theoretical model mean (4,000).")
        print("> Since the owners claimed costs were HIGHER, a sample mean of 3,050 completely contradicts their claim right from the start!")




In [6]:
perform_hypothesis_testing()

STEP 1: HYPOTHESIS STATEMENT
Null Hypothesis (H0): mu <= 4000 (The weekly operating cost matches the theoretical model)
Alternative Hypothesis (H1): mu > 4000 (The weekly operating costs are significantly higher)

STEP 2: CALCULATING THE TEST STATISTIC
Theoretical Mean Weekly Cost (mu): 4000
Standard Deviation of Cost (sigma): 125
Standard Error (SE): 25.0
Calculated Test Statistic: -38.0000

STEP 3: DETERMINING THE CRITICAL VALUE
Significance Level (alpha): 0.05
Critical Value (Z_alpha): 1.6449

STEP 4: MAKING A DECISION
Is -38.0000 > 1.6449?
Decision: Fail to Reject the Null Hypothesis (H0).

STEP 5: CONCLUSION
Conclusion: There is insufficient evidence to support the restaurant owners' claim.
The observed data does not statistically prove that costs have increased beyond the model's prediction.


The Hypotheses (H0 and H1): 
Null Hypothesis (H0): mU<=4000. The weekly operating costs are what the model predicts or lower
Alternative Hypothesis (H1): Mu > 4000. The weekly operating costs have increased.
**we fail to reject the null hypothesis as the Test Statistic: -38.0000(p) >Critical Value- 1.6449

